### PR TITLE
DB: Remove `ErrAlreadyDefined` sentinel error.

### PR DESCRIPTION
**Pull request by markylaing**
This error is imported by the `lxd/request` package so that it canconvert it to an `http.StatusConflict` on a call to `response.SmartError`. Other applications are using the `lxd/request` package. Removing this error prevents the import of `lxd/db` into `lxd/request` and therefore prevents the import of `lxd/db` into downstream repositories.

Unblocks https://github.com/canonical/microcluster/pull/97

### CODE REVIEW

File: lxd/backup.go [hunk 1]
```diff
@@ -57,7 +58,7 @@ func backupCreate(s *state.State, args db.InstanceBackup, sourceInst instance.In
 		return tx.CreateInstanceBackup(ctx, args)
 	})
 	if err != nil {
-		if err == db.ErrAlreadyDefined {
+		if api.StatusErrorCheck(err, http.StatusConflict) {
```
Comment (tomponline) [inline lxd/backup.go:61]: We can just return the err wrapped with something like `Failed creating backup record: %w`?

Comment (markylaing) [reply lxd/backup.go]: Done.

File: lxd/backup.go [hunk 2]
```diff
@@ -410,7 +411,7 @@ func volumeBackupCreate(s *state.State, args db.StoragePoolVolumeBackup, project
 		return tx.CreateStoragePoolVolumeBackup(ctx, args)
 	})
 	if err != nil {
-		if err == db.ErrAlreadyDefined {
+		if api.StatusErrorCheck(err, http.StatusConflict) {
```
[the remark this answers is not in the document]
Comment (markylaing) [reply lxd/backup.go]: Done.

File: lxd/backup.go [hunk 3]
```diff
@@ -57,11 +57,7 @@ func backupCreate(s *state.State, args db.InstanceBackup, sourceInst instance.In
 		return tx.CreateInstanceBackup(ctx, args)
 	})
 	if err != nil {
-		if err == db.ErrAlreadyDefined {
-			return fmt.Errorf("Backup %q already exists", args.Name)
-		}
-
-		return fmt.Errorf("Insert backup info into database: %w", err)
+		return fmt.Errorf("Failed created instance backup record: %w", err)
```
Comment (tomponline) [inline lxd/backup.go:60]: Failed creating... not created

Comment (markylaing) [reply lxd/backup.go:60]: Whoopsie :facepalm:

Comment (markylaing) [reply lxd/backup.go]: Fixed.

**Comment by markylaing** [comment]
Now failing unit tests. Missed some references.

**Review by tomponline — APPROVED** [review]

**Pull request merged**